python/cudf/cudf/core/reshape.py

```diff
@@ -822,22 +822,31 @@ def as_tuple(x):
 
     for v in df:
         names = [as_tuple(v) + as_tuple(name) for name in column_labels]
-        col = df._data[v]
-        result.update(
-            cudf.DataFrame._from_table(
-                col.scatter_to_table(
```

Comment:
Why did we remove the `scatter_to_table` method and expand it's logic here?

Reply:
When I was replacing instances of `_from_table` with `_from_data` I found that the function was typically used in one of three ways. The first is getting data from libcudf, which was being turned into a `Table` that then had `_from_data` called on it. The second was methods that would do something like `_from_table(Frame(data))`. The first case was the main pattern I wanted to eliminate here, while the second is a trivial replacement of a completely unnecessary round trip `dict->Frame->[DataFrame|Series|Index]`. 

The third case was pretty much only happening in this `scatter_to_table`: creating a `Table` subclass inside a pure Python/Cython API that was then having `_from_table` called on it. This code is emblematic of another pattern we want to remove, `Column` APIs that generate `Table`s, which is a circular dependency that we should remove. In this case it's not 100% related to this PR, but since I had to make a change anyway for getting rid of `_from_table` I went ahead and removed this Column method because `scatter_to_table` was only used in this one place. Fwiw `scatter_to_table` was also doing a lot of more general argument handling (e.g. the `nrows`/`ncols` logic) that we don't need since this is the only use case, so the code is more readable this way.